src/npc_attack.cpp

```diff
@@ -32,20 +28,42 @@ const int base_throw_now = 10'000;
 
 // TODO: make a better, more generic "check if this projectile is blocked" function
 // TODO: put this in a namespace for reuse
-static bool has_obstruction( const tripoint &from, const tripoint &to )
+static bool has_obstruction( const tripoint &from, const tripoint &to, bool check_ally = false )
 {
     std::vector<tripoint> line = line_to( from, to );
     // @to is what we want to hit. we don't need to check for obstruction there.
     line.pop_back();
     const map &here = get_map();
     for( const tripoint &line_point : line ) {
-        if( here.impassable( line_point ) ) {
+        if( here.impassable( line_point ) || ( check_ally && g->critter_at( line_point ) ) ) {
             return true;
         }
     }
     return false;
 }
 
+static bool can_move( const npc &source )
+{
+    return !source.in_vehicle && source.rules.engagement != combat_engagement::NO_MOVE;
+}
+
+static bool can_move_melee( const npc &source )
+{
+    return can_move( source ) && source.rules.engagement != combat_engagement::FREE_FIRE;
+}
+
+// TODO: Uncomment when distance checks to allies are added. Clang-tidy doesn't like dead code
+/*static bool enemy_distance_to_ally( const npc &source, const Creature &target )
+{
+    int distance = rl_dist( target.pos(), source.pos() );
+    for( const weak_ptr_fast<Creature> &buddy : source.get_cached_friends() ) {
+        if( !buddy.expired() ) {
+            distance = std::min( distance, rl_dist( target.pos(), buddy.lock()->pos() ) );
+        }
+    }
+    return distance;
+}*/
+
```

Comment:
we don't like keeping commented code in the codebase either. we can just write it again later, but it's best to just remove it if you would keep it commented.

Reply:
```suggestion
```